### ISSUE TITLE
Uploader battery % seems to be getting stuck

### ISSUE THREAD

**Issue by jasoncalabrese** [nightscout/android-uploader#149]
[FB thread](https://www.facebook.com/groups/cgminthecloud/permalink/719378834850042/) about this issue, there are at least 2 people that are able to reproduce this.

**Comment by ELUTE** [comment]
I have seen it MANY times. Work around is to clear device status collection and reboot uploader.

**Comment by ktind** [comment]
What upload mechanism is being used when you all see this? Which version(s) seem to be affected the most?

A future date for the status collection seems to be the most likely culprit - I've seen this once before and a targeted deletion of the offending records resolved the issue. What is the date on the record when you see this issue (seems that part was truncated in the FB post)?

**Comment by jasoncalabrese** [comment]
In Amy's case it was using the rest uploader and there was never any future data.  I know the code for getting the battery level is using some statics, almost seemed like it just kept sending an old value.

**Comment by jasoncalabrese** [comment]
data from mongo show time while battery was stuck at 98%

Over 9 hours at 98%: 2015-02-23T04:20:08.248Z - 2015-02-23T13:45:06.816Z

```
{ "uploaderBattery" : 100, "created_at" : "2015-02-23T14:13:37.741Z", "_id" : ObjectId("54eb3591d23603340339d603") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:45:06.816Z", "_id" : ObjectId("54eb2ee2d23603340339d602") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:40:06.397Z", "_id" : ObjectId("54eb2db6d23603340339d601") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:35:05.968Z", "_id" : ObjectId("54eb2c89d23603340339d600") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:30:06.538Z", "_id" : ObjectId("54eb2b5ed23603340339d5ff") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:25:07.089Z", "_id" : ObjectId("54eb2a33d23603340339d5fe") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:20:06.628Z", "_id" : ObjectId("54eb2906d23603340339d5fd") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:15:06.932Z", "_id" : ObjectId("54eb27dad23603340339d5fc") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:10:06.429Z", "_id" : ObjectId("54eb26aed23603340339d5fb") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:05:06.040Z", "_id" : ObjectId("54eb2582d23603340339d5fa") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T13:00:06.572Z", "_id" : ObjectId("54eb2456d23603340339d5f9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:55:06.098Z", "_id" : ObjectId("54eb232ad23603340339d5f8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:50:06.621Z", "_id" : ObjectId("54eb21fed23603340339d5f7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:45:06.181Z", "_id" : ObjectId("54eb20d2d23603340339d5f6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:40:06.661Z", "_id" : ObjectId("54eb1fa6d23603340339d5f5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:35:06.172Z", "_id" : ObjectId("54eb1e7ad23603340339d5f4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:30:06.671Z", "_id" : ObjectId("54eb1d4ed23603340339d5f3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:25:06.242Z", "_id" : ObjectId("54eb1c22d23603340339d5f2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:20:06.783Z", "_id" : ObjectId("54eb1af6d23603340339d5f1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:15:06.322Z", "_id" : ObjectId("54eb19cad23603340339d5f0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:10:06.873Z", "_id" : ObjectId("54eb189ed23603340339d5ef") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:05:06.425Z", "_id" : ObjectId("54eb1772d23603340339d5ee") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T12:00:05.994Z", "_id" : ObjectId("54eb1645d23603340339d5ed") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:55:06.544Z", "_id" : ObjectId("54eb151ad23603340339d5ec") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:50:06.118Z", "_id" : ObjectId("54eb13eed23603340339d5eb") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:45:06.665Z", "_id" : ObjectId("54eb12c2d23603340339d5ea") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:40:06.205Z", "_id" : ObjectId("54eb1196d23603340339d5e9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:35:06.717Z", "_id" : ObjectId("54eb106ad23603340339d5e8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:30:06.275Z", "_id" : ObjectId("54eb0f3ed23603340339d5e7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:25:06.798Z", "_id" : ObjectId("54eb0e12d23603340339d5e6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:20:06.277Z", "_id" : ObjectId("54eb0ce6d23603340339d5e5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:15:06.776Z", "_id" : ObjectId("54eb0bbad23603340339d5e4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:10:06.308Z", "_id" : ObjectId("54eb0a8ed23603340339d5e3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:05:06.838Z", "_id" : ObjectId("54eb0962d23603340339d5e2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T11:00:06.338Z", "_id" : ObjectId("54eb0836d23603340339d5e1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:55:06.819Z", "_id" : ObjectId("54eb070ad23603340339d5e0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:50:06.349Z", "_id" : ObjectId("54eb05ded23603340339d5df") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:45:06.809Z", "_id" : ObjectId("54eb04b2d23603340339d5de") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:40:06.339Z", "_id" : ObjectId("54eb0386d23603340339d5dd") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:35:06.881Z", "_id" : ObjectId("54eb025ad23603340339d5dc") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:30:06.393Z", "_id" : ObjectId("54eb012ed23603340339d5db") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:25:06.930Z", "_id" : ObjectId("54eb0002d23603340339d5da") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:20:06.472Z", "_id" : ObjectId("54eafed6d23603340339d5d9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:15:06.873Z", "_id" : ObjectId("54eafdaad23603340339d5d8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:10:07.392Z", "_id" : ObjectId("54eafc7fd23603340339d5d7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:05:06.924Z", "_id" : ObjectId("54eafb52d23603340339d5d6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T10:00:06.434Z", "_id" : ObjectId("54eafa26d23603340339d5d5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:55:07.964Z", "_id" : ObjectId("54eaf8fbd23603340339d5d4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:50:06.427Z", "_id" : ObjectId("54eaf7ced23603340339d5d3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:45:06.897Z", "_id" : ObjectId("54eaf6a2d23603340339d5d2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:40:06.346Z", "_id" : ObjectId("54eaf576d23603340339d5d1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:35:06.796Z", "_id" : ObjectId("54eaf44ad23603340339d5d0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:30:06.197Z", "_id" : ObjectId("54eaf31ed23603340339d5cf") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:25:06.646Z", "_id" : ObjectId("54eaf1f2d23603340339d5ce") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:20:07.156Z", "_id" : ObjectId("54eaf0c7d23603340339d5cd") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:15:06.611Z", "_id" : ObjectId("54eaef9ad23603340339d5cc") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:10:07.029Z", "_id" : ObjectId("54eaee6fd23603340339d5cb") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:05:06.398Z", "_id" : ObjectId("54eaed42d23603340339d5ca") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T09:00:06.790Z", "_id" : ObjectId("54eaec16d23603340339d5c9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:55:06.210Z", "_id" : ObjectId("54eaeaead23603340339d5c8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:50:06.690Z", "_id" : ObjectId("54eae9bed23603340339d5c7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:45:17.112Z", "_id" : ObjectId("54eae89dd23603340339d5c6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:40:13.529Z", "_id" : ObjectId("54eae76dd23603340339d5c5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:35:10.003Z", "_id" : ObjectId("54eae63ed23603340339d5c4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:30:06.494Z", "_id" : ObjectId("54eae50ed23603340339d5c3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:25:07.024Z", "_id" : ObjectId("54eae3e3d23603340339d5c2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:20:06.452Z", "_id" : ObjectId("54eae2b6d23603340339d5c1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:15:06.963Z", "_id" : ObjectId("54eae18ad23603340339d5c0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:10:06.423Z", "_id" : ObjectId("54eae05ed23603340339d5bf") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:05:06.856Z", "_id" : ObjectId("54eadf32d23603340339d5be") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T08:00:06.307Z", "_id" : ObjectId("54eade06d23603340339d5bd") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:55:06.797Z", "_id" : ObjectId("54eadcdad23603340339d5bc") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:50:06.287Z", "_id" : ObjectId("54eadbaed23603340339d5bb") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:45:06.758Z", "_id" : ObjectId("54eada82d23603340339d5ba") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:40:06.247Z", "_id" : ObjectId("54ead956d23603340339d5b9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:35:06.708Z", "_id" : ObjectId("54ead82ad23603340339d5b8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:30:06.158Z", "_id" : ObjectId("54ead6fed23603340339d5b7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:25:06.640Z", "_id" : ObjectId("54ead5d2d23603340339d5b6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:20:07.139Z", "_id" : ObjectId("54ead4a7d23603340339d5b5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:15:06.599Z", "_id" : ObjectId("54ead37ad23603340339d5b4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:10:07.112Z", "_id" : ObjectId("54ead24fd23603340339d5b3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:05:06.593Z", "_id" : ObjectId("54ead122d23603340339d5b2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T07:00:09.102Z", "_id" : ObjectId("54eacff9d23603340339d5b1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:55:08.494Z", "_id" : ObjectId("54eaceccd23603340339d5b0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:50:08.082Z", "_id" : ObjectId("54eacda0d23603340339d5af") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:45:09.225Z", "_id" : ObjectId("54eacc75d23603340339d5ae") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:40:07.767Z", "_id" : ObjectId("54eacb47d23603340339d5ad") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:35:07.874Z", "_id" : ObjectId("54eaca1bd23603340339d5ac") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:30:06.527Z", "_id" : ObjectId("54eac8eed23603340339d5ab") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:25:07.036Z", "_id" : ObjectId("54eac7c3d23603340339d5aa") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:20:06.480Z", "_id" : ObjectId("54eac696d23603340339d5a9") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:15:06.908Z", "_id" : ObjectId("54eac56ad23603340339d5a8") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:10:06.417Z", "_id" : ObjectId("54eac43ed23603340339d5a7") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:05:06.918Z", "_id" : ObjectId("54eac312d23603340339d5a6") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T06:00:06.389Z", "_id" : ObjectId("54eac1e6d23603340339d5a5") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:55:24.880Z", "_id" : ObjectId("54eac0ccd23603340339d5a4") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:50:21.321Z", "_id" : ObjectId("54eabf9dd23603340339d5a3") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:45:17.748Z", "_id" : ObjectId("54eabe6dd23603340339d5a2") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:40:14.290Z", "_id" : ObjectId("54eabd3ed23603340339d5a1") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:35:10.780Z", "_id" : ObjectId("54eabc0ed23603340339d5a0") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:30:07.193Z", "_id" : ObjectId("54eabadfd23603340339d59f") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:25:07.623Z", "_id" : ObjectId("54eab9b3d23603340339d59e") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:20:08.054Z", "_id" : ObjectId("54eab888d23603340339d59d") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:15:08.445Z", "_id" : ObjectId("54eab75cd23603340339d59c") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:10:08.723Z", "_id" : ObjectId("54eab630d23603340339d59b") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:05:07.650Z", "_id" : ObjectId("54eab503d23603340339d59a") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T05:00:06.835Z", "_id" : ObjectId("54eab3d6d23603340339d599") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:55:08.244Z", "_id" : ObjectId("54eab2acd23603340339d598") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:50:06.652Z", "_id" : ObjectId("54eab17ed23603340339d597") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:45:07.085Z", "_id" : ObjectId("54eab053d23603340339d596") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:40:07.464Z", "_id" : ObjectId("54eaaf27d23603340339d595") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:35:06.686Z", "_id" : ObjectId("54eaadfad23603340339d594") }
{ "uploaderBattery" : 98, "created_at" : "2015-02-23T04:32:21.147Z", "_id" : ObjectId("54eaad55d23603340339d593") }
{ "uploaderBattery" : 99, "created_at" : "2015-02-23T04:20:08.248Z", "_id" : ObjectId("54eaaa78d23603340339d592") }
```

**Comment by ktind** [comment]
I suspect this was fixed with #135 and #140. I'll send out a test build to a few users to verify.

**Comment by ktind** [comment]
Fixed in the latest alpha
